### PR TITLE
target/riscv: move read redirection for `priv` to `riscv-013.c`

### PR DESCRIPTION
**Pull request by en-sc**
The reason for the change is a conflict: `dcsr[5]` is `dcsr.v` in current spec, but it is `dcsr.debugint` in 0.11. This causes `priv` register to be read incorrectly.

Change-Id: If2d8fdcd8536afa4c7149c453101b00ce0df1ce0

### CODE REVIEW

**Comment by en-sc** [comment]
@lz-bro, FYI.

**Comment by TommyMurphyTM1234** [comment]
Confirmed as working with SiFive HiFive1 Rev A01:

* https://github.com/riscv/riscv-openocd/issues/996#issuecomment-1896039265

**Comment by JanMatCodasip** [comment]
FYI, I will probably be able to review this on Monday.

File: src/target/riscv/riscv-013.c [hunk 1]
```diff
@@ -4851,6 +4851,19 @@ struct target_type riscv013_target = {
 static int riscv013_get_register(struct target *target,
 		riscv_reg_t *value, enum gdb_regno rid)
 {
+	/* This would be nice to move this redirection, to version-independent part,
```
Comment (MarekVCodasip) [inline src/target/riscv/riscv-013.c]: I would reword this:
```suggestion
	/* It would be beneficial to move this redirection to the version-independent section,
```

**Review by MarekVCodasip — CHANGES_REQUESTED** [review]
Otherwise, it looks fine to me.

File: src/target/riscv/riscv-013.c (hunk 1 above)
Comment (en-sc) [reply src/target/riscv/riscv-013.c]: Addressed.

Comment (MarekVCodasip) [reply src/target/riscv/riscv-013.c]: Thanks.

**Review by MarekVCodasip — APPROVED** [review]

**Comment by JanMatCodasip** [comment]
Let's merge this on Thursday if no other feedback comes.

**Comment by JanMatCodasip** [comment]
> TommyMurphyTM1234: Confirmed as working with SiFive HiFive1 Rev A01: [...]

Thanks for checking it. :+1:

**Comment by lz-bro** [comment]
Need to move write register for priv  to riscv-013.c?

**Comment by en-sc** [comment]
> Need to move write register for priv to riscv-013.c?

@iz-bro, this is a valid concern. However, I've decided not to move the write in this commit.
The reasoning is:
* This is a bug-fix. Bug is related to reading the register, not writing it.
* Assuming the other code is correct, only a user-requested write can cause an issue with writing `dcsr[5]` via writing `priv` on 0.11.

I'm currently working on a patch that significantly changes register accesses. I will post a PR changing the code in question soon. So, I would like not to change current version too much to avoid doing extra work).
